src/XCCDF/result.c

```diff
@@ -261,25 +278,67 @@ void xccdf_result_fill_sysinfo(struct xccdf_result *result)
 	struct xccdf_target_fact *fact = NULL;
 	const char *probe_root = getenv("OSCAP_PROBE_ROOT");
 
-#ifdef OSCAP_UNIX
 	_xccdf_result_clear_metadata(XITEM(result));
+	_xccdf_result_fill_scanner(result);
+
+#ifdef OSCAP_UNIX
 	char *hostname = NULL;
+	char *fqdn = NULL;
 	if (probe_root) {
 		hostname = _get_etc_hostname(probe_root);
 	} else {
 		struct utsname sname;
 		if (uname(&sname) != -1)
 			hostname = strdup(sname.nodename);
+
+		if (hostname) {
+			struct addrinfo hints, *info, *p;
+			int gai_res;
+
+			memset(&hints, 0, sizeof hints);
+			hints.ai_family = AF_UNSPEC;
+			hints.ai_socktype = SOCK_STREAM;
+			hints.ai_flags = AI_CANONNAME;
+
+			if (!(gai_res = getaddrinfo(hostname, NULL, &hints, &info))) {
+				for(p = info; p != NULL; p = p->ai_next) {
+					fact = xccdf_target_fact_new();
+					xccdf_target_fact_set_name(fact, "urn:xccdf:fact:asset:identifier:fqdn");
+					xccdf_target_fact_set_string(fact, p->ai_canonname);
+					/* store FQDN under XCCDF1.2 (6.6.3) predefined name */
+					_xccdf_result_add_target_fact_uniq(result, fact);
+					if (!fqdn)
+						fqdn = strdup(p->ai_canonname);
+				}
+				freeaddrinfo(info);
+			} else {
+				dI("Unable to get FQDN(s) via getaddrinfo: %s\n", gai_strerror(gai_res));
```

Comment:
The `dI` macro will print end of line automatically.

```suggestion
				dI("Unable to get FQDN(s) via getaddrinfo: %s", gai_strerror(gai_res));
```

Reply:
Yeah, missed that one

Reply:
Fixed